cdap-proto/src/main/java/co/cask/cdap/proto/id/SecureKeyId.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.id;
+
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.element.EntityType;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import javax.ws.rs.BadRequestException;
+
+/**
+ * Uniquely identifies a Secure store key.
+ */
+public class SecureKeyId extends EntityId implements NamespacedId, ParentedId<NamespaceId> {
+  // KMS only supports lower case keys.
+  private static final Pattern secureKeyNamePattern = Pattern.compile("[a-z0-9_-]+");
+
+  private final String namespace;
+  private final String name;
+  private transient Integer hashCode;
+
+  public SecureKeyId(String namespace, String name) {
+    super(EntityType.SECUREKEY);
+    if (!isValidSecureKey(name)) {
+      throw new BadRequestException("Improperly formatted secure key name. The name can contain lower case " +
```

Comment:
wait, this is unchecked? That's awesome! every ``EntityId`` class should do this then.